lib/graceful-exit.js

```diff
@@ -80,8 +80,10 @@ exports.gracefulExitHandler = function(app, server, _options) {
 
   // Disconnect all the socket.io clients
   if (options.socketio) {
-    options.socketio.sockets.clients().forEach(function(socket) {
-      logger('Killing socketio socket');
+    var connectedSockets = Object.values(options.socketio.sockets.sockets);
+    logger('Disconnecting ' + connectedSockets.length + ' sockets...');
+    options.socketio.close();
+    connectedSockets.forEach(function(socket) {
```

Comment:
Alex, thank you for submitting a patch.

Could you give me some background on what changed with socket.io v1.4 that requires this change? And is it backwards compatible?  I scanned the socket.io change blog and didn't see anything about it.